pkg/segment/writer/segstore.go

```diff
@@ -823,8 +848,37 @@ func (segstore *SegStore) computeStarTree() {
 			segstore.usingSegTree = false
 			return
 		}
+
+		hasTreeSpace := false
+		atreeCounterLock.Lock()
+		if currentAgileTreeCount < MaxConcurrentAgileTrees {
+			// for now the first MaxConcurrentAgileTrees segstores will get to AgileTree but
+			// we should add some smart logic on how we can rotate this
+			// amongst other indices/segstores
+			currentAgileTreeCount++
+			hasTreeSpace = true
+		}
+		atreeCounterLock.Unlock()
+		if !hasTreeSpace {
+			segstore.usingSegTree = false
+			return
+		}
+
 		segstore.usingSegTree = true
-		segstore.sbuilder.ResetSegTree(&segstore.wipBlock, sortedGrpCols, mCols)
+		sizeToAdd := len(sortedGrpCols) - len(segstore.stbDictEncWorkBuf)
+		if sizeToAdd > 0 {
+			newArr := make([][]string, sizeToAdd)
+			segstore.stbDictEncWorkBuf = append(segstore.stbDictEncWorkBuf, newArr...)
+		}
+		for colNum := 0; colNum < len(sortedGrpCols); colNum++ {
+			if len(segstore.stbDictEncWorkBuf[colNum]) < MaxAgileTreeNodeCount {
+				// we know each stree col won't have more encodings than max stree node limit
+				segstore.stbDictEncWorkBuf[colNum] = make([]string, MaxAgileTreeNodeCount)
+			}
+		}
```

Comment:
nit: I'm a little confused by this comment, because you only know that since the `if` condition was true. So this comment isn't useful? It looks like you might have moved this logic along with the comment, and now that it's here the comment isn't needed